src/Mailbox.php

```diff
@@ -181,11 +181,14 @@ public function getMessages(ConditionInterface $search = null, int $sortCriteria
             }
             $messageNumbers = \imap_search(...$params);
         }
+        $error = \imap_last_error();
+        if (false !== $error) {
+            // this way all errors occurred during search will be reported
+            throw new InvalidSearchCriteriaException(
+                \sprintf('Invalid search criteria [%s] - %s', $query, $error)
```

Comment:
There is no need for saving the `imap_last_error` output and putting it into the exception message: exceptions already output all the error stack

Reply:
Sorry, already fixed that in a [new commit](https://github.com/ddeboer/imap/pull/445/commits/06bffa2298be4fc88583bb7b7909ae4d32aa951e)